mlflow/types/schema.py

```diff
@@ -408,3 +408,236 @@ def __eq__(self, other) -> bool:
 
     def __repr__(self) -> str:
         return repr(self.inputs)
+
+
+@experimental
+class ParamSpec:
+    """
+    Specification used to represent parameters for the model.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        type: Union[DataType, str],  # pylint: disable=redefined-builtin
```

Comment:
How would we extend this if there is a request to add support for dict?

Reply:
Can we extend this to Union[DataType, dict[str, DataType], str]?

Reply:
Yes I don't think it'll be necessary though. Each dict key can be pulled into a separate Param

Reply:
Parameter types we might struggle to support in the future:

```python
# A list of tuples with different types
p = [("1", 1), ("2", 2)]
```

```python
# List of list, each inner list has different number of elements
p = [[1], [1, 2], [1, 2, 3]]
```

```python
# List of dict
p = [{"a": 0}, {"a": 1}]
```

Reply:
I think that's okay for now - thanks @harupy  :)